packages/apps/plugins/plugin-markdown/src/MarkdownPlugin.tsx

```diff
@@ -244,135 +218,54 @@ export const MarkdownPlugin = (): PluginDefinition<MarkdownPluginProvides> => {
       },
       surface: {
         component: ({ data, role, ...props }, forwardedRef) => {
-          // TODO(wittjosiah): Ideally this should only be called when the editor is actually being used.
-          //   We probably want a better pattern for splitting this surface resolver up.
           const doc =
             data.active instanceof DocumentType
               ? data.active
               : data.object instanceof DocumentType
                 ? data.object
                 : undefined;
-          const space = doc && getSpace(doc);
-          const extensions = useMemo(() => {
-            // TODO(wittjosiah): Autocomplete is not working and this query is causing performance issues.
-            // const query = space?.db.query(Filter.schema(DocumentType));
-            // query?.subscribe();
-            return getCustomExtensions(doc /*, query */);
-          }, [doc, space, settings.values.editorInputMode, getViewMode(doc?.id)]);
-
-          const dispatch = useIntentDispatcher();
-          const handleCommentSelect = useCallback(() => {
-            void dispatch({ action: LayoutAction.SET_LAYOUT, data: { element: 'complementary', state: true } });
-          }, [dispatch]);
-
-          const fileManagerPlugin = useResolvePlugin(parseFileManagerPlugin);
-          const handleFileUpload = useMemo(() => {
-            if (space === undefined) {
-              return undefined;
-            }
-
-            if (fileManagerPlugin?.provides.file.upload === undefined) {
-              return undefined;
-            }
-
-            return async (file: File) => {
-              return fileManagerPlugin?.provides?.file?.upload?.(file, space);
-            };
-          }, [fileManagerPlugin, space]);
-
-          const handleViewModeChange = useCallback(
-            (viewMode: EditorViewMode) => {
-              if (doc) {
-                state.values.viewMode[doc.id] = viewMode;
-              }
-            },
-            [doc?.id],
-          );
 
           switch (role) {
-            // TODO(burdon): Normalize layout (reduce variants).
+            case 'section':
             case 'article': {
-              if (doc) {
+              if (doc && doc.content) {
                 return (
                   <DocumentEditor
-                    viewMode={getViewMode(doc.id)}
-                    toolbar={settings.values.toolbar}
                     document={doc}
-                    extensions={extensions}
-                    onCommentSelect={handleCommentSelect}
-                    onFileUpload={handleFileUpload}
-                    onViewModeChange={handleViewModeChange}
+                    extensions={state.values.extensionProviders.reduce((acc: Extension[], provider) => {
```

Comment:
this isn't memoized, is that going to be a problem?

Reply:
Maybe… I waffled back and forth about passing `extensionProviders` directly. I may just do that.

Reply:
Memoized now, though it’s only handled by `DocumentEditor` — should the upstream `MarkdownEditor` do this instead and filter by a possibly falsy `document`?

Reply:
I think that's likely necessary. Unfortunately comments is somewhat a required extension at the moment because it's still a little coupled so that needs to always be available.